Ukoly/DU_10/generator.py

```diff
@@ -0,0 +1,118 @@
+#
+# české názvy jsou ponechány po vzoru zadání
+
+from datetime import datetime, timedelta
+import requests
+import random
+import json
+import qrcode
+now = datetime.now()
+url = (f'https://data.kurzy.cz/json/meny/b[2]den[{now.strftime("%Y%m%d")}].json')
+
+
+class CurrencyError(Exception):
+    def __init__(self, name):
+        self.name = name
+
+    def __str__(self):
+        return(f'Tato měna není na seznamu')
+
+
+class LoadExchangeRates():
+    def __init__(self, name):
+        self.name = name
+
+    def load_data(self):
+        exchange = requests.get(url=url)
+        with open("kurz_cnb.json", 'wb') as text_file:
+            text_file.write(exchange.content)
+
+    def exchange_rates(self):
+        with open("kurz_cnb.json", 'r') as text_file:
+            data = json.load(text_file)
+            json_date = datetime.strptime(data["den"], "%Y%m%d")
+            if now - json_date > timedelta(days=1):
+                LoadExchangeRates.load_data(self)
```

Comment:
Toto nie je správne, nakoľko aktuálne si pristúpila k danej metóde ak k statickej. Ty chceš pristupovať priamo z instancie, teda správne `self.load_data()`, kedy premenná `self` obsahuje danú instanciu. To, že `self` nedávaš ako argument je úplne v poriadku, za teba to vyplní interpreter už spomínanou instanciou na ktorej si to zavolala...

Toto je dosť potrebné aby si pochopila... takže sa pýtaj.